### PR TITLE
WP-9043 - support more types

### PR DESCRIPTION
**Pull request by yi-varicent**

### CODE REVIEW

File: tap_redshift/__init__.py [hunk 1]
```diff
@@ -170,13 +175,18 @@ def schema_for_column(c):
...
+        result.format = 'symon.geo'
+
     else:
         result = Schema(None,
                         inclusion='unsupported',
                         description='Unsupported column type {}'
                         .format(column_type))
 
-    if column_nullable == 'yes':
+    if column_nullable == 'yes' and result.type != None:
```
Comment (kchan-varicent) [inline tap_redshift/__init__.py]: `is not None`

File: tap_redshift/__init__.py [hunk 2]
```diff
@@ -47,6 +47,9 @@
 DATETIME_TYPES = {'timestamp', 'timestamptz',
                   'timestamp without time zone', 'timestamp with time zone'}
 
+GEOMETRY = {'geometry'}
+SELECT_FORMAT = { 'symon.geo': 'ST_AsEWKT("{}")' }
```
Comment (kchan-varicent) [inline tap_redshift/__init__.py:51]: what does this mean? :/

File: tap_redshift/__init__.py [hunk 3]
```diff
@@ -170,13 +175,18 @@ def schema_for_column(c):
         result.type = 'string'
         result.format = 'date'
 
+    elif column_type in GEOMETRY:
+        LOGGER.info("in other types")
```
Comment (kchan-varicent) [inline tap_redshift/__init__.py]: what does this log mean?

File: tap_redshift/__init__.py [hunk 4]
```diff
@@ -268,6 +278,8 @@ def row_to_record(catalog_entry, version, row, columns, time_extracted):
 
 
 def sync_table(connection, catalog_entry, state):
+    LOGGER.info(str(catalog_entry.schema))
+    LOGGER.info(catalog_entry.schema)
```
Comment (kchan-varicent) [inline tap_redshift/__init__.py]: double debug?

Comment (yi-varicent) [reply tap_redshift/__init__.py]: lol

File: tap_redshift/__init__.py (hunk 2 above)
Comment (yi-varicent) [reply tap_redshift/__init__.py:51]: normally its some unreadable string, this makes it string readable "POINT(1 2)" or whatever other format like "POLYGON(...)"
https://postgis.net/docs/ST_AsEWKT.html

File: tap_redshift/__init__.py (hunk 1 above)
Comment (yi-varicent) [reply tap_redshift/__init__.py]: also updated this so it sets ["null"] always

File: tap_redshift/__init__.py [hunk 5]
```diff
@@ -170,14 +173,21 @@ def schema_for_column(c):
...
                         inclusion='unsupported',
                         description='Unsupported column type {}'
                         .format(column_type))
 
     if column_nullable == 'yes':
-        result.type = ['null', result.type]
+        if result.type is not None:
+            result.type = ['null', result.type]
+        else:
+            result.type = ['null']
```
Comment (yi-varicent) [inline tap_redshift/__init__.py:190]: updated to always have null

File: tap_redshift/__init__.py (hunk 2 above)
Comment (yi-varicent) [reply tap_redshift/__init__.py:51]: image

**Review by kchan-varicent — APPROVED** [review]

**Pull request merged**